### PR TITLE
NCI-Agency/anet#1029: Correctly show sub-navigation in all cases

### PR DESCRIPTION
**Pull request by gjvoosten**
Fixes NCI-Agency/anet#1029

### CODE REVIEW

File: client/src/components/SubNav.js [hunk 1]
```diff
@@ -12,17 +12,25 @@ export default class SubNav extends Component {
...
 
 		this.state = {
-			subnavElem: document.getElementById(this.props.subnavElemId),
+			subnavElem: null,
 		}
 	}
 
-	componentDidMount() {
+	updateSubnav = () => {
 		const elem = document.getElementById(this.props.subnavElemId)
```
Comment (VassilIordanov) [inline client/src/components/SubNav.js]: Can we use https://reactjs.org/docs/refs-and-the-dom.html instead of `getElementById`?

Comment (gjvoosten) [reply client/src/components/SubNav.js]: According to https://reactjs.org/docs/portals.html the second arg to `createPortal` must be a DOM node.

Comment (VassilIordanov) [reply client/src/components/SubNav.js]: `.current`?

Comment (gjvoosten) [reply client/src/components/SubNav.js]: Re: your first remark; changing it to just this (plus `propTypes`):
```
	render() {
		return (
			ReactDOM.createPortal(
				this.props.children,
				document.getElementById(this.props.subnavElemId)
			)
		)
	}
```
could probably work.

Comment (gjvoosten) [reply client/src/components/SubNav.js]: But I get an occasional `Error: Target container is not a DOM element.` when trying that.

Comment (gjvoosten) [reply client/src/components/SubNav.js]:
```
	render() {
		const subnavElem = document.getElementById(this.props.subnavElemId)
		return subnavElem && ReactDOM.createPortal(
			this.props.children,
			subnavElem
		)
	}
```
works much better!

**Pull request merged**